packages/react/src/profiler.tsx

```diff
@@ -0,0 +1,122 @@
+import { getCurrentHub } from '@sentry/browser';
+import { Integration, IntegrationClass } from '@sentry/types';
+import { logger } from '@sentry/utils';
+import * as hoistNonReactStatic from 'hoist-non-react-statics';
+import * as React from 'react';
+
+export const UNKNOWN_COMPONENT = 'unknown';
+
+const TRACING_GETTER = ({
+  id: 'Tracing',
+} as any) as IntegrationClass<Integration>;
+
+/**
+ *
+ * Based on implementation from Preact:
+ * https:github.com/preactjs/preact/blob/9a422017fec6dab287c77c3aef63c7b2fef0c7e1/hooks/src/index.js#L301-L313
+ *
+ * Schedule a callback to be invoked after the browser has a chance to paint a new frame.
+ * Do this by combining requestAnimationFrame (rAF) + setTimeout to invoke a callback after
+ * the next browser frame.
+ *
+ * Also, schedule a timeout in parallel to the the rAF to ensure the callback is invoked
+ * even if RAF doesn't fire (for example if the browser tab is not visible)
+ *
+ * This is what we use to tell if a component activity has finished
+ *
+ */
+function afterNextFrame(callback: Function): void {
+  let timeout: number | undefined;
+  let raf: number;
+
+  const done = () => {
+    window.clearTimeout(timeout);
+    window.cancelAnimationFrame(raf);
+    window.setTimeout(callback);
+  };
+
+  raf = window.requestAnimationFrame(done);
+  timeout = window.setTimeout(done, 100);
+}
+
+const getInitActivity = (componentDisplayName: string): number | null => {
+  const tracingIntegration = getCurrentHub().getIntegration(TRACING_GETTER);
+
+  if (tracingIntegration !== null) {
+    // tslint:disable-next-line:no-unsafe-any
+    const activity = (tracingIntegration as any).constructor.pushActivity(componentDisplayName, {
+      data: {},
+      description: `<${componentDisplayName}>`,
+      op: 'react',
+    });
+
+    // tslint:disable-next-line: no-unsafe-any
+    return activity;
+  }
+
+  logger.warn(`Unable to profile component ${componentDisplayName} due to invalid Tracing Integration`);
```

Comment:
```suggestion
  logger.warn(`Unable to profile component ${componentDisplayName} due to invalid Tracing Integration.`);
  logger.warn(`Please make sure to setup the Tracing integration.`);
```
^ A more guiding warning for the user.